### PR TITLE
Projections proto definition change.

### PR DESCRIPTION
**Pull request by YegorUdovchenko**
In this PR:

- Added identifiers for non singleton projections.
- Completed documentation in `projections.proto`.
- Extracted month type to the `MonthlySpendingsReportId`.
- Removed some redundant fields and comments in `projections.proto`.

### CODE REVIEW

File: model/src/main/proto/javaclasses/mealorder/identifiers.proto [hunk 1]
```diff
@@ -111,3 +110,41 @@ message PurchaseOrderId {
     // A date on which this purchase order is formed.
     google.protobuf.Timestamp po_date = 2;
 }
+
+// An ID of a menu list projection for UI role User.
+//
+message MenuListId {
+
+    // Date on which the menu list is formed.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/identifiers.proto]: A date on which the menu list is formed.

File: model/src/main/proto/javaclasses/mealorder/identifiers.proto [hunk 2]
```diff
@@ -111,3 +110,41 @@ message PurchaseOrderId {
...
+    spine.time.LocalDate date = 1;
+}
+
+// An ID of a order list projection for UI role User.
+//
+message OrderListId {
+
+    // Date on which the order list is formed.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/identifiers.proto]: A date on which the menu list is formed.

File: model/src/main/proto/javaclasses/mealorder/identifiers.proto [hunk 3]
```diff
@@ -111,3 +110,41 @@ message PurchaseOrderId {
...
+
+    // An identifier of a user who made orders.
+    UserId user_id = 2;
+}
+
+// An ID of a monthly spendings report projection for UI role Admin.
+//
+message MonthlySpendingsReportId {
+
+    // Month on which the spendings are calculated.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/identifiers.proto]: A month on which the spendings are calculated.

File: model/src/main/proto/javaclasses/mealorder/identifiers.proto [hunk 4]
```diff
@@ -111,3 +110,41 @@ message PurchaseOrderId {
...
+    LocalMonth month = 1;
+
+    // Representation of a month of year.
+    //
+    message LocalMonth {
+
+        // Month value.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/identifiers.proto]: A month value.

File: model/src/main/proto/javaclasses/mealorder/identifiers.proto [hunk 5]
```diff
@@ -111,3 +110,41 @@ message PurchaseOrderId {
...
+        spine.time.MonthOfYear month = 1;
+
+        // Year value.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/identifiers.proto]: A year value.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 6]
```diff
@@ -37,42 +37,39 @@ import "spine/net/email_address.proto";
...
 
     // Vendor name.
     VendorName vendor_name = 1;
 
-    //TODO: Maybe it should be replaced with `MenuDateRange` to prevent `MenuItem` duplication for each date with same dish list.
-    // Date on which the menu is attached.
-    spine.time.LocalDate date = 2;
-
     // Collection of dishes.
     repeated DishItem dishes = 3;
```
Comment (vladislav-kozachenko) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: `repeated DishItem dishes = 2;`

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 7]
```diff
@@ -111,8 +108,8 @@ message MenuCalendarView {
 
 // Item of menu calendar.
 //
-// Item state indicates that some active
-// attached menus exist for this date or not.
+// Item state indicates wether some active
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: Item state indicates whether there is attached menu for this day or not.

ps: wether -это кастрированный баран.

Comment (vladislav-kozachenko) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // Item state indicates **whether** some active

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 8]
```diff
@@ -287,26 +285,32 @@ enum PurchaseOrderStatus {
...
+    PurchaseOrderStatus purchase_order_status = 2;
 
     // Collection of dishes.
-    repeated DishItem dishes = 2;
+    repeated DishItem dishes = 3;
 }
 
 // A projection state of purchase order details view
-// sorted by User.
+// sorted by user.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: You don't need it here.
"sorted by user" should be over the field `orders .`

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 9]
```diff
@@ -37,42 +37,39 @@ import "spine/net/email_address.proto";
...
-// **** Projections for UI role User ****
+// **** Projections for the UI role User. ****
 
-// A projection state of menu list for the specified date.
+// A projection state of a menu list for the specified date.
 //
-// This view includes list of actual attached menus for the specified date.
-// Menu is attached if its date range includes specified date.
+// This view includes the list of actual attached menus for the specified date.
+// Date is specified by `MenuListId` value.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: The date is specified by `MenuListId` value.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto (hunk 8 above)
Comment (vladislav-kozachenko) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: Maybe 'grouped' instead 'sorted' might be better.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 10]
```diff
@@ -99,7 +96,7 @@ message DishItem {
 
 // A projection state of menu calendar.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // A projection state of a menu calendar.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 11]
```diff
@@ -172,8 +172,6 @@ message VendorListView {
 
 // Item of vendor list.
 //
-// This item contains vendor identifier, name, email and phone numbers.
-//
 message VendorItem {
 
     // An identifier of the vendor.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // An identifier of a vendor.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 12]
```diff
@@ -250,7 +248,7 @@ message PurchaseOrderListView {
 // When administrator decides to cancel purchase order
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // When administrator decides to cancel a purchase order

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 13]
```diff
@@ -250,7 +248,7 @@ message PurchaseOrderListView {
...
 // in some reason its status gets `CANCELED` value.
 //
-// When administrator has received ordered dishes and checked
+// When administrator receives ordered dishes and checks
 // the accordance to the purchase order dish list, he should
 // mark purchase order as delivered. After this purchase order
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // mark the purchase order as delivered. After this purchase order

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 14]
```diff
@@ -287,26 +285,32 @@ enum PurchaseOrderStatus {
 //
 message PurchaseOrderDetailsByDishView {
 
-    // Purchase order on which the details view is formed.
-    PurchaseOrderItem purchase_order = 1;
+    // An identifier of the purchase order.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // An identifier of a purchase order.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 15]
```diff
@@ -287,26 +285,32 @@ enum PurchaseOrderStatus {
...
+    PurchaseOrderId id = 1;
+
+    // Purchase order status.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // A purchase order status.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 16]
```diff
@@ -287,26 +285,32 @@ enum PurchaseOrderStatus {
...
+// sorted by user.
 //
 // When purchase order is created its validation may complete with failure.
 // Those orders which caused that failure are marked as invalid.
 //
 message PurchaseOrderDetailsByUserView {
 
-    // Purchase order on which the details view is formed.
-    PurchaseOrderItem purchase_order = 1;
+    // An identifier of the purchase order.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // An identifier of a purchase order.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 17]
```diff
@@ -287,26 +285,32 @@ enum PurchaseOrderStatus {
...
+    PurchaseOrderId id = 1;
+
+    // Purchase order status.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // A purchase order status.

File: model/src/main/proto/javaclasses/mealorder/q/projections.proto [hunk 18]
```diff
@@ -330,52 +334,31 @@ message UserOrderDetails {
...
 // This view contains user spendings for full month period
 // if the report is formed for the passed month.
 // It contains user spendings for period from  month first
 // date till today if the report is formed for the current month.
 //
 message MonthlySpendingsReportView {
 
-    // Date range on which the spendings are calculated.
-    LocalMonth report_month = 1;
+    // An identifier of the monthly spendings report.
```
Comment (YuriiHaidamaka) [inline model/src/main/proto/javaclasses/mealorder/q/projections.proto]: // An identifier of monthly spendings report.

**Comment by YegorUdovchenko** [comment]
Made necessary edits.

**Pull request merged**